### ISSUE TITLE
[BXT] Finding tables instead of methods

### ISSUE THREAD

**Issue by playermove** [YaLTeR/BunnymodXT#281]
# Introduction
Less reversing and simple. I'll show you an example.

## Variables
```cpp
cldll_func_t* pClient, Client;
```

## Hooking
```cpp
pClient = FindClient();

if (!pClient)
    return NULL;

Client = *pClient

if (!Client.V_CalcRefdef)
    return NULL;

pClient->CL_CreateMove = Hooked_CL_CreateMove;
```

## Unhooking
```cpp
if (!pClient)
    return NULL;

*pClient = Client; pClient = NULL;
```

**Comment by YaLTeR** [comment]
This is a valid approach, I actually used it a little for the server [in bxt-rs](https://github.com/YaLTeR/bxt-rs/blob/a92dab202aac4dbeb16171bf985c9bf511503a77/src/hooks/server.rs#L29) and was planning to use it for the client as well eventually. However, it requires finding and hooking the server and client loading and unloading functions to do it properly, so it's not always easier than just finding the functions directly. In BXT itself I just historically used pattern search for all functions, and a lot of the functions hooked nowadays are engine functions anyway that aren't present in any tables.

**Comment by playermove** [comment]
> a lot of the functions hooked nowadays are engine functions anyway that aren't present in any tables.

Yeah a lot of them aren't in any tables, but you use some functions from these tables:

**Client**
> cldll_func_t

**Engine**
> cl_enginefunc_t

**Studio**
> engine_studio_api_t

And more ...
